core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -83,51 +83,11 @@ private[spark] class CoarseGrainedExecutorBackend(
     }(ThreadUtils.sameThread)
   }
 
-  // Check that the actual resources discovered will satisfy the user specified
-  // requirements and that they match the configs specified by the user to catch
-  // mismatches between what the user requested and what resource manager gave or
-  // what the discovery script found.
-  private def checkResourcesMeetRequirements(
-      resourceConfigPrefix: String,
-      reqResourcesAndCounts: Array[(String, String)],
-      actualResources: Map[String, ResourceInformation]): Unit = {
-
-    reqResourcesAndCounts.foreach { case (rName, reqCount) =>
-      if (actualResources.contains(rName)) {
-        val resourceInfo = actualResources(rName)
-
-        if (resourceInfo.addresses.size < reqCount.toLong) {
-          throw new SparkException(s"Resource: $rName with addresses: " +
-            s"${resourceInfo.addresses.mkString(",")} doesn't meet the " +
-            s"requirements of needing $reqCount of them")
-        }
-        // also make sure the resource count on start matches the
-        // resource configs specified by user
-        val userCountConfigName =
-          resourceConfigPrefix + rName + SPARK_RESOURCE_COUNT_POSTFIX
-        val userConfigCount = env.conf.getOption(userCountConfigName).
-          getOrElse(throw new SparkException(s"Resource: $rName not specified " +
-            s"via config: $userCountConfigName, but required, " +
-            "please fix your configuration"))
-
-        if (userConfigCount.toLong > resourceInfo.addresses.size) {
-          throw new SparkException(s"Resource: $rName, with addresses: " +
-            s"${resourceInfo.addresses.mkString(",")} " +
-            s"is less than what the user requested for count: $userConfigCount, " +
-            s"via $userCountConfigName")
-        }
-      } else {
-        throw new SparkException(s"Executor resource config missing required task resource: $rName")
-      }
-    }
-  }
-
   // visible for testing
   def parseOrFindResources(resourcesFile: Option[String]): Map[String, ResourceInformation] = {
     // only parse the resources if a task requires them
-    val taskResourceConfigs = env.conf.getAllWithPrefix(SPARK_TASK_RESOURCE_PREFIX)
-    val resourceInfo = if (taskResourceConfigs.nonEmpty) {
-      val execResources = resourcesFile.map { resourceFileStr => {
+    val resourceInfo = if (env.conf.getAllWithPrefix(SPARK_TASK_RESOURCE_PREFIX).nonEmpty) {
```

Comment:
`SPARK_EXECUTOR_RESOURCE_PREFIX`?

Reply:
No, I only want to look for resources is a task we are running requires them.  If no tasks require other resources, then its just overhead for us.